openpype/hosts/houdini/api/shelves.py

```diff
@@ -0,0 +1,200 @@
+import os
+import logging
+import platform
+
+from openpype.settings import get_project_settings
+
+import hou
+
+log = logging.getLogger("openpype.hosts.houdini.shelves")
+
+
+def generate_shelves():
+    """This function generates complete shelves from shelf set to tools
+    in Houdini from openpype project settings houdini shelf definition.
+
+    Raises:
+        FileNotFoundError: Raised when the shelf set filepath does not exist
+    """
+    current_os = platform.system().lower()
+
+    # load configuration of houdini shelves
+    project_settings = get_project_settings(os.getenv("AVALON_PROJECT"))
+    shelves_set_config = project_settings["houdini"]["shelves"]
+
+    if not shelves_set_config:
+        log.debug(
+            "No custom shelves found in project settings."
+        )
+        return
+
+    for shelf_set_config in shelves_set_config:
+        shelf_set_filepath = shelf_set_config.get('shelf_set_source_path')
+
+        if shelf_set_filepath[current_os]:
+            if not os.path.isfile(shelf_set_filepath[current_os]):
+                raise FileNotFoundError(
```

Comment:
`FileNotFoundError` is not global in Python 2, so it fails with `NameError: global name 'FileNotFoundError' is not defined` in Houdini 18.5 with py2.

Might be patched with this added somewhere at the top of the file:

```python
import six
if six.PY2: FileNotFoundError = IOError
```